src/instructlab/process/process.py

```diff
@@ -0,0 +1,296 @@
+                encoding="utf-8",
+                bufsize=1,  # Line-buffered for real-time output
+            )
+            time.sleep(1)
+            # we need to get all of the children processes spawned
+            # to be safe, we will need to try and kill all of the ones which still exist when the user wants us to
+            # however, representing this to the user is difficult. So let's track the parent pid and associate the children with it in the registry
+            max_retries = 5
+            retry_interval = 0.5  # seconds
+            parent = psutil.Process(p.pid)
+            for _ in range(max_retries):
+                children = parent.children(recursive=True)
+                if children:
+                    for child in children:
+                        children_pids.append(child.pid)
+                    break
+                time.sleep(retry_interval)
+            else:
+                logger.warning("No child processes detected. Tracking parent process.")
+            pid = p.pid
+            # Check if subprocess was successfully started
+            if p.poll() is not None:
+                logger.warning(f"Process {p.pid} failed to start.")
+                return None  # Process didn't start
+
+    # Add the process info to the shared registry
+    process_registry[str(local_uuid)] = {
+        "pid": pid,
+        "children_pids": children_pids,
+        "type": process_type,
+        "log_file": log_file,
+        "start_time": datetime.strptime(
+            start_time_str, "%Y-%m-%d %H:%M:%S"
+        ).isoformat(),
+    }
+    logger.info(
+        f"Started subprocess with PID {pid}. Logs are being written to {log_file}."
+    )
+    save_registry(
+        process_registry=process_registry
+    )  # Persist registry after adding process
+    if process_mode == ILAB_PROCESS_MODES.ATTACHED:
+        with open(log_file, "a+") as log:
+            sys.stdout = Tee(log)
+            sys.stderr = sys.stdout
+            try:
+                target(**kwargs)  # Call the function
+            finally:
+                # Restore the original stdout and stderr after the function completes
+                sys.stdout = sys.__stdout__
+                sys.stderr = sys.__stderr__
+                process_registry.pop(local_uuid, None)
+                save_registry(process_registry=process_registry)
+
+
+def list_processes():
+    process_registry = load_registry()
+    if not process_registry:
+        logger.info("No processes currently in the registry.")
+        return
+
+    list_of_processes = []
+    for local_uuid, entry in process_registry.items():
+        now = datetime.now()
+
+        # Calculate runtime
+        runtime = now - datetime.fromisoformat(entry.get("start_time"))
+        # Convert timedelta to a human-readable string (HH:MM:SS)
+        total_seconds = int(runtime.total_seconds())
+        hours, remainder = divmod(total_seconds, 3600)
+        minutes, seconds = divmod(remainder, 60)
```

Comment:
This could probably be its own helper function 
```suggestion
        runtime_str = str(runtime).split(".")[0]
```

Reply:
I am confused what you're suggesting here? You want me to remove the hr, min, seconds calc but leave the below setting of runtime_str?

Reply:
Also this chunk of code is only used in one spot, so I don't see the benefit of making it a function since its 3 lines

Reply:
fair enough